FastExcel/DefinedName.cs

```diff
@@ -0,0 +1,201 @@
+            }
+            catch (Exception exception)
+            {
+                throw new DefinedNameLoadException("Unable to open stream to read internal workbook.xml file", exception);
+            }
+            if (document == null)
+            {
+                throw new DefinedNameLoadException("Unable to load workbook.xml from open stream. Probable corrupt file.");
+            }
+
+            var definedNames = new Dictionary<string, DefinedName>();
+
+            foreach (var e in (from d2 in document.Descendants().Where(dx => dx.Name.LocalName == "definedNames").Descendants()
+                               select d2))
+            {
+                var currentDefinedName = new DefinedName(e);
+                definedNames.Add(currentDefinedName.Key, currentDefinedName);
+            }
+            DefinedNames = definedNames;
+        }
+
+        /// <summary>
+        /// Retrieves ranges of cells by their defined name
+        /// </summary>
+        /// <param name="definedName">Defined Name</param>
+        /// <param name="worksheetIndex">If scoped to a sheet, the worksheetIndex</param>
+        /// <returns>List of cells encapsulated in another list representing seperate ranges</returns>
+        public IEnumerable<IEnumerable<Cell>> GetCellRangesByDefinedName(string definedName, int? worksheetIndex = null)
+        {
+            List<List<Cell>> result = new List<List<Cell>>();
+
+            string key = (!worksheetIndex.HasValue ? definedName : definedName + ":" + worksheetIndex);
+
+            if (!DefinedNames.ContainsKey(key))
+                return result;
+
+            string[] references = DefinedNames[key].Reference.Split(',');
+
+            foreach(string reference in references)
+            {
+                // If A) is a formula or B) does not contain ! or $ then this reference is not supported
+                if (Regex.IsMatch(reference, @"[\(\)\*\+\-\/]") || !Regex.IsMatch(reference, @"[!$]"))
+                    continue;
+
+                CellRange cellRange = new CellRange(reference);
+
+                Worksheet worksheet = Read(cellRange.SheetName);
+
+                result.Add(worksheet.GetCellsInRange(cellRange).ToList());
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// Gets all cells by defined name
+        /// Like GetCellRangesByCellName, but just retreives all cells in a single list
+        /// </summary>
+        /// <param name="definedName"></param>
+        /// <param name="worksheetIndex"></param>
+        /// <returns></returns>
+        public IEnumerable<Cell> GetCellsByDefinedName(string definedName, int? worksheetIndex = null)
+        {
+            var cells = new List<Cell>();
+            var cellRanges = GetCellRangesByDefinedName(definedName) as List<List<Cell>>;
+
+            foreach (var cellRange in cellRanges)
+                cells.InsertRange(cells.Count, (from cell in cellRange select cell));
```

Comment:
Redundant linq statement, should be: `cells.InsertRange(cells.Count, cellRange);